### ISSUE TITLE
[Test failure] TC_LEARNER_50: Missing search box in the learner dashboard

### ISSUE THREAD

**Issue by angonz** [openedx/wg-build-test-release#431]
### Release

Sumac

### Expected behavior

There should be a search box in the learner dashboard.

### Actual behavior

There is no search box in the learner dashboard.

### Steps to reproduce

Go to the LMS, learner dashboard.
![image](https://github.com/user-attachments/assets/16864893-c4fc-4102-88d4-ee296580d956)

**Comment by magajh** [comment]
Thanks for the report, I've confirmed this issue is occurring for both the [community remote environment](https://apps.sandbox.openedx.org/learner-dashboard/) and the sandbox for Sumac

**Comment by crathbun428** [comment]
@magajh and @angonz - I went back and looked at the learner dashboard in Redwood, and it looks the same as it does for Sumac. 

Yeah this is an old test that needs to be reworked - thank you for pointing this out. There is a search bar on both, but it only appears if you have permissions to masquerade (or view content as X user). See the screenshots of Redwood and Sumac below.
![Screenshot 2024-12-05 at 9 15 25 AM](https://github.com/user-attachments/assets/5cef0596-6f3f-4baa-8b55-0efde1480baf)
![Screenshot 2024-12-05 at 9 15 16 AM](https://github.com/user-attachments/assets/87c7746b-81a3-4ed3-96e7-f92957b12a08)
 

If you're logged in as an active learner, you wouldn't see it. I wouldn't consider this a test failure knowing this - I can update the test now.

**Comment by angonz** [comment]
I think that the test refers to searching between the student's courses for course content, not the user selector for masquerading. This feature [was available](https://github.com/openedx/edx-platform/blob/0bd0e6f4cab4d398216f010b6cd09ab21e7297e2/lms/templates/courseware/courses.html#L39-L54) before the learner-dashboard MFE and AFAIK there is no DEPR ticket for it.

**Comment by angonz** [comment]
I'm asking to reopen for a clarification on this point.

**Comment by crathbun428** [comment]
@angonz - Thank you for this context - let me see if I can tap someone who has more experience with this particular page to weigh in.

**Comment by crathbun428** [comment]
@angonz - I checked in with an FE engineer here and he couldn't recall there being a search bar on this page either. A brief look at the repo turned up nothing as well. Could you provide details and screenshots of an environment where this search bar exists? Trying to figure out where this search bar was lost (if possible) to help get to the bottom of this. 

cc: @brian-smith-tcril

**Comment by magajh** [comment]
@crathbun428 In case it's helpful to have this information, the old learner dashboard did include a search bar for courses. 
![legacy-learner-dashboard](https://github.com/user-attachments/assets/cc51e15a-81ec-4013-8eda-83d113f0e8e5)

There's even an old ticket for this same test related to the previous dashboard during the Quince release: https://github.com/openedx/wg-build-test-release/issues/321

What I initially thought is that the Learner Dashboard MFE doesn’t actually have a search bar, and this test was likely never removed when the MFE became the default dashboard. In fact, the test sheet shows that the test also failed in Redwood.

I opened this issue hoping someone could help confirm this, but I haven’t received any response yet: https://github.com/openedx/frontend-app-learner-dashboard/issues/521

**Comment by crathbun428** [comment]
Excellent - thank you, @magajh. It looks like the search bar was potentially just removed when this was converted to an MFE. I do think it makes sense to have one on this page.

**Comment by angonz** [comment]
I think this is something that should be discussed as part of the product agenda. We have received requests from customers to support a course search functionality. I believe it's a useful feature allowing students to look for something they remember having seen somewhere but not exactly where.

**Comment by crathbun428** [comment]
@mariajgrimaldi - I'm now reminded that I asked Jenna about this a while back - it seems this page was recently (ahead of Redwood completely redesigned by PMs over at 2U and it appears that the decision was made to remove the search bar from this page - let me reach out to the PM over at 2U to figure out why - I'd expect that had some user-backed reasoning for this - this will help us inform why it was removed/should it be brought back, etc). cc: @angonz

**Comment by mariajgrimaldi** [comment]
Hi, @crathbun428, thank you! Do we have any updates on this? Were you able to reach out to 2U's PM?

**Comment by crathbun428** [comment]
Yes - the PM didn't know if this was intentional. We'll need a small product proposal to add this back (since this is an FE change - happy to draft it up).

**Comment by mariajgrimaldi** [comment]
Let's focus the efforts in the https://github.com/openedx/platform-roadmap/issues/443 issue so we don't duplicate info. Thank you all!
